src/test/java/com/loofah/graph/api/integrationTests/IntegrationTest.java

```diff
@@ -67,51 +75,69 @@ public void setUp() {
+
+        JsonNode parsedResponseBody = objectMapper.readTree(response.getBody()).get(DATA).get(SKILLS);
         List<LinkedHashMap> allSkills = objectMapper.convertValue(parsedResponseBody, ArrayList.class);
 
         assertEquals(HttpStatus.OK, response.getStatusCode());
         assertEquals(3, allSkills.size());
     }
 
     @Test
-    public void returns_all_seeded_categories() throws IOException {
+    public void returns_correct_category_when_id_is_valid() throws IOException {
 
-        Request request = getRequest(allCategoriesQuery);
+        ResponseEntity<String> response = callAPI(categoryQuery);
 
-        final HttpEntity<Object> requestEntity = new HttpEntity<>(request, headers);
+        JsonNode parsedResponseBody = objectMapper.readTree(response.getBody()).get(DATA).get(CATEGORY);
+        LinkedHashMap selectedSkill = objectMapper.convertValue(parsedResponseBody, LinkedHashMap.class);
 
-        ResponseEntity<String> response = testRestTemplate.exchange(getURI(), HttpMethod.POST, requestEntity, String.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals("categoryDescription1", selectedSkill.get("description"));
+    }
+
+    @Test
+    public void returns_all_seeded_categories() throws IOException {
+
+        ResponseEntity<String> response = callAPI(categoriesQuery);
 
-        JsonNode parsedResponseBody = objectMapper.readTree(response.getBody()).get(DATA).get(ALL_CATEGORIES);
+        JsonNode parsedResponseBody = objectMapper.readTree(response.getBody()).get(DATA).get(CATEGORIES);
         List<LinkedHashMap> allCategoriesResponse = objectMapper.convertValue(parsedResponseBody, ArrayList.class);
 
         assertEquals(HttpStatus.OK, response.getStatusCode());
         assertEquals(2, allCategoriesResponse.size());
     }
 
     @Test
-    public void returns_correct_skill_when_id_is_valid() throws IOException {
-
-        Request request = getRequest(skillQueryValidID);
+    public void returns_correct_skills_for_category_id() throws IOException {
 
-        final HttpEntity<Object> requestEntity = new HttpEntity<>(request, headers);
-
-        ResponseEntity<String> response = testRestTemplate.exchange(getURI(), HttpMethod.POST, requestEntity, String.class);
+        ResponseEntity<String> response = callAPI(skillsByCategoryQuery);
 
-        JsonNode parsedResponseBody = objectMapper.readTree(response.getBody()).get(DATA).get(SKILL);
-        LinkedHashMap selectedSkill = objectMapper.convertValue(parsedResponseBody, LinkedHashMap.class);
+        JsonNode parsedResponseBody = objectMapper.readTree(response.getBody()).get(DATA).get(SKILLS_BY_CATEGORY);
+        List<LinkedHashMap> selectedSkillsForCategory = objectMapper.convertValue(parsedResponseBody, ArrayList.class);
 
         assertEquals(HttpStatus.OK, response.getStatusCode());
-        assertEquals("description1", selectedSkill.get("description"));
+        assertEquals(2, selectedSkillsForCategory.size());
+    }
```

Comment:
Nitpick: might be worth also verifying that the two skills returned have the expected category ID?

Reply:
yes good shout, updated (but we should think about these tests in general, they feel very sort of brittle to me)